apps/mobile/src/components/Button.tsx

```diff
@@ -1,32 +1,34 @@
 import clsx from 'clsx';
 import { ReactNode } from 'react';
-import { ActivityIndicator, TouchableOpacity, TouchableOpacityProps, View } from 'react-native';
+import { ActivityIndicator, View } from 'react-native';
 
+import { GalleryTouchableOpacity, GalleryTouchableOpacityProps } from './GalleryTouchableOpacity';
 import { Typography } from './Typography';
 
 type Variant = 'primary' | 'secondary';
 
 type ButtonProps = {
+  style?: GalleryTouchableOpacityProps['style'];
   className?: string;
-  style?: TouchableOpacityProps['style'];
   loading?: boolean;
   disabled?: boolean;
   text: string;
   icon?: ReactNode;
   variant?: Variant;
-  onPress?: TouchableOpacityProps['onPress'];
-};
+  trackProperties?: Record<string, unknown>;
```

Comment:
do we need this prop if the intersection already has `properties`? If I understand this correctly, now `Button` takes both `properties` and `trackProperties`